controller/orphan_controller.go

```diff
@@ -261,15 +261,30 @@ func (oc *OrphanController) reconcile(orphanName string) (err error) {
 		log.Infof("Orphan got new owner %v", oc.controllerID)
 	}
 
+	existingOrphan := orphan.DeepCopy()
+
 	if !orphan.DeletionTimestamp.IsZero() {
+		defer func() {
+			if reflect.DeepEqual(existingOrphan.Status, orphan.Status) {
+				return
+			}
+			_, updateStatusErr := oc.ds.UpdateOrphanStatus(orphan)
+			if updateStatusErr != nil {
+				log.WithError(updateStatusErr).Errorf("Failed to update condition while cleaning up %v orphan %v", orphan.Spec.Type, orphan.Name)
+				if err == nil {
+					err = updateStatusErr
+				}
```

Comment:
Might not need to pass `updateStatusErr` to `err`. The condition update is best-effort. Logging the error is good enough. WDYT?